### PR TITLE
Decorators

### PR DESCRIPTION
**Pull request by pinkmoon25**
### school library: decorate a class

- Used the Decorator design pattern to correct the names of people
- Created Nameable class
- Created BaseDecorator class
- Created CapitalizeDecorator and TrimmerDecorator

### CODE REVIEW

File: ruby/decorator.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+require './nameable'
+require './person'
```
Comment (GzimAsani) [inline ruby/decorator.rb]: - [x] Change all the `require` to `require_relative`. It is a common practice to use relative when doing it for other classes which are present on your project

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello @PinkMoon25   :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your project  :+1:  
 
 ## Required Changes 

- [ ] Check the comment below :arrow_double_down: 






 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: ruby/decorator.rb (hunk 1 above)
Comment (pinkmoon25) [reply ruby/decorator.rb]: hi @GzimAsani I have made the changes, although I think those are optional changes

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @PinkMoon25 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by pinkmoon25** [comment]
Thanks, @youmari and @GzimAsani  for the review :+1:

**Pull request merged**